x-pack/plugins/security_solution/server/lib/detection_engine/signals/build_events_query.ts

```diff
@@ -43,36 +43,62 @@ export const buildEventsSearchQuery = ({
     format: 'strict_date_optional_time',
   }));
 
-  const sortField =
-    timestampOverride != null && !excludeDocsWithTimestampOverride
-      ? timestampOverride
-      : '@timestamp';
+  const rangeFilter: estypes.QueryContainer[] =
+    timestampOverride != null
+      ? [
+          {
+            range: {
+              [timestampOverride]: {
+                lte: to,
+                gte: from,
+                format: 'strict_date_optional_time',
+              },
+            },
+          },
+          {
+            bool: {
+              filter: [
+                {
+                  range: {
+                    '@timestamp': {
+                      lte: to,
+                      gte: from,
+                      // @ts-expect-error
+                      format: 'strict_date_optional_time',
```

Comment:
typscript complains that the `format` field is not applicable to the `@timestamp` field. I think the type is wrong.